x-pack/libbeat/management/managerV2.go

```diff
@@ -231,6 +228,16 @@ func NewV2AgentManagerWithClient(config *Config, registry *reload.Registry, agen
 // Beats central management interface implementation
 // ================================
 
+func (cm *BeatV2Manager) AgentInfo() client.AgentInfo {
+	// on normal/production flow this should not happen. However, it was panicking
+	// on some integration tests.
```

Comment:
This is a little bit vague and seems like it is unresolved.

This is because the paths that only call `NewV2()` don't get agent info to set until there is a checkin, correct?